src/pyfmi/simulation/assimulo_interface_fmi3.pyx

```diff
@@ -0,0 +1,611 @@
+        self._model.time = start_time
+
+        self.t0 = start_time
+        self.y0 = self._model.continuous_states
+        self.problem_name = self._model.get_name()
+
+        [f_nbr, g_nbr] = self._model.get_ode_sizes()
+
+        self._f_nbr = f_nbr
+        self._g_nbr = g_nbr
+        self._A = None
+
+        # Used by Assimulo
+        self.state_events_use = g_nbr > 0
+        self.time_events_use = True
+
+        # If there is no state in the model, add a dummy state der(y)=0
+        if f_nbr == 0:
+            self.y0 = np.array([0.0])
+
+        # Determine the result file name
+        if result_file_name == '':
+            self.result_file_name = model.get_name()+'_result.txt'
+        else:
+            self.result_file_name = result_file_name
+        self.debug_file_name = model.get_name().replace(".","_")+'_debug.txt'
+        self.debug_file_object = None
+
+        # Default values
+        self.export = result_handler
+
+        # Internal values
+        self._logg_step_event = []
+        self._write_header = True
+        self._logging = logging
+        self._sparse_representation = False
+        # self._with_jacobian = with_jacobian # TODO
+        self._with_jacobian = False
+
+        # # If result handler support is available, logging turns into dynamic_diagnostics
+        self._logging_as_dynamic_diagnostics = self._logging and result_handler.supports.get("dynamic_diagnostics", False)
+        self._number_of_diagnostics_variables = number_of_diagnostics_variables
+
+        self.jac_use = False
+        # if f_nbr > 0 and with_jacobian:
+        #     self.jac_use = True # Activates the jacobian
+
+        #     # Need to calculate the nnz.
+        #     [derv_state_dep, derv_input_dep] = model.get_derivatives_dependencies()
+        #     self.jac_nnz = np.sum([len(derv_state_dep[key]) for key in derv_state_dep.keys()])+f_nbr
+
+        # if extra_equations:
+        #     self._extra_f_nbr = extra_equations.get_size()
+        #     self._extra_y0    = extra_equations.y0
+        #     self.y0 = np.append(self.y0, self._extra_y0)
+        #     self._extra_equations = extra_equations
+
+        #     if hasattr(self._extra_equations, "jac"):
+        #         if hasattr(self._extra_equations, "jac_nnz"):
+        #             self.jac_nnz += extra_equations.jac_nnz
+        #         else:
+        #             self.jac_nnz += len(self._extra_f_nbr)*len(self._extra_f_nbr)
+        # else:
+        #     self._extra_f_nbr = 0
+        
+        if synchronize_simulation:
+            try:
+                if synchronize_simulation is True:
+                    self._synchronize_factor = 1.0
+                elif synchronize_simulation > 0:
+                    self._synchronize_factor = synchronize_simulation
+                else:
+                    raise InvalidOptionException(f"Setting {synchronize_simulation} as 'synchronize_simulation' is not allowed. Must be True/False or greater than 0.")
```

Comment:
Let's move this message up between `if` and `try` and then we can simply `raise ...(msg)` at both line 151 and 153

Reply:
Not sure I understand; can you add the suggestion?

The `else` covers e.g., negative values, the `except:` covers nonsense values where e.g., the `synchronize_simulation > 0` evaluation would cause an exception.

Sidenote: I propose we fix this option to be just a float for FMI v3, the unclear type of this is a mess.

Reply:
I could be more clear, I just mean that we can move the message to a variable to reduce the duplicate exact string, i.e.
```
msg = "..."
if ...
 raise Exception1(msg)
except ...
 raise Exception2(msg)
```
(since we raise the exact same error message)